### PR TITLE
Balder/mbus transported with headers and body in slime

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR
The first commit is what you have review earlier on.
The rest adds more tests for versions and enable the new protocol as of 6.148.
It also corrects a bug in c++ version selection logic.

### CODE REVIEW

File: messagebus/src/tests/sendadapter/sendadapter.cpp [hunk 1]
```diff
@@ -251,3 +190,38 @@ Test::testVersionedSend(TestData &data,
...
+    EXPECT_TRUE(dynamic_cast<mbus::RPCSendV1 *>(data._srcServer.net.getSendAdapter(vespalib::Version(5, 0))) != nullptr);
+    EXPECT_TRUE(data._srcServer.net.getSendAdapter(vespalib::Version(6, 147)) != nullptr);
+    EXPECT_TRUE(dynamic_cast<mbus::RPCSendV1 *>(data._srcServer.net.getSendAdapter(vespalib::Version(6, 147))) != nullptr);
+    EXPECT_TRUE(data._srcServer.net.getSendAdapter(vespalib::Version(6, 148)) != nullptr);
+    EXPECT_TRUE(dynamic_cast<mbus::RPCSendV2 *>(data._srcServer.net.getSendAdapter(vespalib::Version(6, 148))) != nullptr);
+    EXPECT_TRUE(data._srcServer.net.getSendAdapter(vespalib::Version(9, 999)) != nullptr);
+    EXPECT_TRUE(dynamic_cast<mbus::RPCSendV2 *>(data._srcServer.net.getSendAdapter(vespalib::Version(9, 999))) != nullptr);
+}
+
+TEST("test that ee can send between multiple versions") {
```
Comment (vekterli) [inline messagebus/src/tests/sendadapter/sendadapter.cpp]: s/ee/we

**Review by vekterli — APPROVED** [review]
:+1: let's get this show on the road.

@havardpe feel free to have a final glance at this as well.

**Comment by vekterli** [comment]
By the way, our internal version number has been bumped to 6.148.x already. Presumably this won't introduce conflicts as long as none of those internal versions are promoted to release candidates.

**Comment by baldersheim** [comment]
I will change this to 149, to make sure we get this tested during upgrade of canary in integration tests.

**Pull request merged**